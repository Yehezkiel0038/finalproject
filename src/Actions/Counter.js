import ActionTypes from '../Actions/Index'

export const increaseCount = () => ({
     type: ActionTypes.INCREASE_COUNT
})  

export const decreaseCount = () => ({
     type: ActionTypes.DECREASE_COUNT
})