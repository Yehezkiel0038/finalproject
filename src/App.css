.App {
  text-align: center;
}
a:visited{
  color: #26dac2;
}

.header {
  background-color: #f1f1f1;
  padding: 20px;
  text-align: center;
}

a:hover{
  color: rgb(255, 0, 55);
}
.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.navbar {
  overflow: hidden;
  background-color: #2c3952;
}
.navbar a:hover {
  background-color: #ddd;
  color: black;
}

.navbar a {
  float: center;
  display: block;
  color: #ffffff;
  text-align: center;
  padding: 16px 18px;
  text-decoration: none;
  background-color: #2c3952;
}
.App-header {
  background-color: #2c3952;
  min-height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: #efefef;
}

.App-link {
  color: #66fb61;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

