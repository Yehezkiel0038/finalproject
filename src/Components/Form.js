import React, {Component} from 'react'

class Form extends Component{
  
  constructor(props){
        super(props);
        this.state = {
          name:'',
          income:'',
          work:'',
          skills:[],
          address:''
        }
          this.onChange= this.onChange.bind(this);
          this.onSubmit= this.onSubmit.bind(this);
          this.onPick= this.onPick.bind(this);
          this.onValueChange= this.onValueChange.bind(this);
      }
      onSubmit(event){
        event.preventDefault();
        event.stopPropagation();
        console.log(this.state)
      }
      onChange(event){
        const{ name, value} = event.target;
        this.setState({[name]:value})
      }
      onPick(event){
        let skills =[...this.state.skills];
        let index =skills.findIndex(elemen=> event.target.value === elemen);
        if (index>0){
          skills= [...skills.slice(0,index), ...skills.slice(index + 1, skills.length)]
        }else if(index === 0){
          skills= [...skills.slice(index+1, skills.length)]
        }else{
          skills.push(event.target.value)
        }
        console.log(skills);
        this.setState({skills});
      }
      onValueChange(event) {
        this.setState({
          sex: event.target.value
        });
      }
    
      render(){
        const{name, income, skills,work, address}=this.state;
    
        return(
          
          <form className="App-header" onSubmit={this.onSubmit}>
            <h1>Form Daftar Kerja</h1>
            <label>
              Nama : 
              <input name="name" type="text" value={name} onChange={this.onChange} />
            </label>
            <br/>
            <label>
              Penghasilan yang diharapkan : 
              <input name="income" type="number" value={income} onChange={this.onChange} />
            </label>
            <br/>
            <label>
              Tanggal Mulai Kerja : 
              <input name="work" type="date" value={work} onChange={this.onChange} />
            </label>
            <br/>
            <label>Jenis Kelamin : </label>
            <div className="radio">
              <label>
                <input
                  type="radio" value="male"
                  checked={this.state.sex === "male"}
                  onChange={this.onValueChange}
                />
                Pria
              </label>
            </div>
            <div className="radio">
              <label>
                <input
                  type="radio" value= "female"
                  checked={this.state.sex === "female"}
                  onChange={this.onValueChange}
                />
                Wanita
              </label>
            </div>
    
            <br/>
            <label>
              Kemampuan :  
              <select multiple={true} value={skills} onChange={this.onPick}>
                <option value="fullstack">JavaScript</option>
                <option value="frontend">HTML</option>
                <option value="backend">Phyton</option>
                <option value="game">C++</option>
                </select>
            </label>
            <br/>
            <label>
              Alamat:
              <textarea name="address" value={address} onChange={this.onChange} />
            </label>
            <br/>
            <button type="submit">Submit</button>
          </form>
        )
      }
    
    }

    export default Form;