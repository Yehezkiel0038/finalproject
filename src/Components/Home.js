import React, {Component} from 'react';
 export  class Home extends Component{
    render(){
      return (<div>
        <div className="App">
          <header className="App-header">
            Halaman Utama
          </header>
       </div>
      </div>)
    } 
   }