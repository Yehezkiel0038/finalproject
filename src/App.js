import React from 'react';
import { createStore } from 'redux'
import { Provider } from 'react-redux'

import Counter from './Components/Counter'
import counterReducer from './Reducers/counterReducer'
import Form from './Components/Form'

const store = createStore(counterReducer)

function App() {
    
  return (
    
    <Provider store={store}>  
       
        <Counter />
        <Form /> 
    </Provider>
  );
}

export default App;
