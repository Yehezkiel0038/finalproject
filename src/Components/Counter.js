import React, {Component} from 'react'

import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { 
    increaseCount,
    decreaseCount
} from '../Actions/Counter'
import '../App.css'
import { 
    Link,
    BrowserRouter as Router,
    Switch,
    Route } from 'react-router-dom'
import {Home} from './Home'
import Form from './Form'

class Counter extends Component {

static mapStateToProps = state => {
    return {
        count: state.count
    }
}

static mapDispatchToProps = dispatch => {
    return bindActionCreators({
        increaseCount,
        decreaseCount
    },
    dispatch
    )
}

componentDidUpdate(){
    console.log("Berhasil Update")
}

render() {
    const { increaseCount, decreaseCount } = this.props
    return(
        
        <Router>
        <h1 className="App-header">Selamat Datang</h1>
             {/* buat route */}
               <ul className="navbar">
               <Link style={{ cursor: 'pointer'}}  to="/" >Home</Link>
               <Link style={{ cursor: 'pointer'}}  to="/Counter">Hitungan</Link>
               <Link style={{ cursor: 'pointer'}}  to="/Form">Form</Link>
               </ul>
            
       
        <Switch>
            <Route path="/">
            <Home />
            </Route>
            <Route path="/Counter">
            <Counter />
            </Route>
            <Route path="/Form">
            <Form />
            </Route>
            
        </Switch>
        <div className="App-header" style={{backgroundColor: "#2c3952",
            minHeight: "100vh",
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
            justifyContent: "center",
            color: "#efefef"
                   
                }}>
           
            <p style={{color:"white"}} >Hitungan : {this.props.count}</p>
        {/* buat button counter */}
        <button  style={{ backgroundColor: "#00bfff"}} onClick={increaseCount} >Tambahkan</button>
        
        <button  style={{ backgroundColor: "#8000ff"}} onClick={decreaseCount}>Kurangkan</button>
      
        </div>
        </Router>
    )
}
    
}
  

export default connect(
    Counter.mapStateToProps,
    Counter.mapDispatchToProps
)(Counter)

