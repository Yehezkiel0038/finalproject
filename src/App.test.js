import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { Home } from './Components/Home';
import renderer from 'react-test-renderer'
import Form from './Components/Form';
import Counter from './Components/Counter';

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('Home', () => {
  const component = renderer.create(
    <Home />,
    <Form />,
    <Counter/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

 
});
