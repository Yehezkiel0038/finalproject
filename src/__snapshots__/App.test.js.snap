// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 1`] = `
<div>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      Halaman Utama
    </header>
  </div>
</div>
`;
